Makefile

```
@@ -184,17 +187,22 @@ eclint/lint:
 	$(ECLINT_FILES) | grep -zv ".bats" | xargs -0 -I {} eclint check {}
 	@ echo "[$@]: Project PASSED eclint test!"
 
-python/%: FIND_PYTHON := find . $(FIND_EXCLUDES) -name '*.py' -type f
-## Lints Python files
-python/lint: | guard/program/black
+python/%: PYTHON_FILES := git ls-files '*.py'
```

One further bit here is that we should catch both committed and untracked files... suggest these options:

```suggestion
python/%: PYTHON_FILES ?= git ls-files --cached --others --exclude-standard '*.py'
```

---

I made a note in Trello to ask whether I should exit if there are files that have not been committed or print a warning.  But I can do this instead.

---

--cached is set by default, so I'm going to omit that from the line unless you see a reason for keeping it.

---

I might have been doing it wrong, or had some local issue, but with only --others I didn't get any committed files in the list.

---

When you say "only --others", are you saying you simply had 'git ls-files --others'?

I checked in a change with 'git ls-files --others --exclude-standard'.

I'm calling it a night .. I've got things that were fixed that are now broken and things that were broken had been working.  It's a mess.

---

yes, here's how i tested...

create a new .py file:
```
$ touch file.py
```

test the default:
```
$ git ls-files '*.py'
pyppyn/__init__.py
pyppyn/cli.py
setup.py
tests/minipippy/setup.py
tests/minipippy/src/minipippy/__init__.py
tests/test_pyppyn.py
```

try --others, note what should be an ignored file and the absence of committed files:
```
$ git ls-files --others '*.py'
file.py
tests/minipippy/tests/minipippy/.pyppyn/dist/temp/minipippy/__init__.py
```

try --others --exclude-standard:
```
$ git ls-files --others --exclude-standard '*.py'
file.py
```

add --cached:
```
$ git ls-files --cached --others --exclude-standard '*.py'
file.py
pyppyn/__init__.py
pyppyn/cli.py
setup.py
tests/minipippy/setup.py
tests/minipippy/src/minipippy/__init__.py
tests/test_pyppyn.py
```

---

All three arguments to 'git ls-files' will be used: --cached --others --exclude-standard.